app/Http/Controllers/V1/ReservationRecurringController.php

```diff
@@ -60,6 +60,19 @@ public function __invoke(ReservationRecurringRequest $request)
         }
     }
 
+    /**
+     * Remove all records by recurring id from storage.
+     *
+     * @param  Reservation  $reservation
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy(Reservation $reservation)
+    {
+        Reservation::where('recurring_id', $reservation->recurring_id)->delete();
```

Comment:
ini bisa seperti ini gak yah kang @samudra-ajri 
```suggestion
        $reservation->delete();
```

Reply:
Punten, Kang @ayocodingit 
Pas saya coba pakai cara tsb hanya delete records pertama saja, Kang.

Reply:
oh tujuannya bisa multiple reccuring id yang sama yah kang @samudra-ajri 
berarti untuk kasus ini tidak perlu menggunakan route model binding

Reply:
Siap, Kang.. :+1:

Reply:
Punten, Kang @ayocodingit 
Klw dia ngga pakai model binding kena `authorizeResource`.

Reply:
kecuali klw `public function __construct()
    {
        $this->authorizeResource(Reservation::class);
    }` kita hapus

Reply:
oalah kena authorizeResource yah 
berarti boleh gunakan teknik model observer kang tapi di taro di model aja contoh nya
![image](https://user-images.githubusercontent.com/41193120/118757309-0c3f9c80-b897-11eb-8a18-92ca51024671.png)
agar tetap single responsibility principle :smile:

Reply:
Hum, saya kira maksud dari _single responsibility principle_ lebih ke prilaku dan objek yg berada dalam satu class saja, Kang? Misalkan priakunya khusus untuk crud saja, atau misalkan khusus untuk menghandel objek user saja. Sedangkan teknis dibebaskan.

Gitu bukan ya, Kang?
cmiiw

Reply:
jadi simple nya mah satu fungsi satu goal :smile:

Reply:
Hho.. mantap, siap, Kang..

Reply:
Punten, Kang @ayocodingit 
Klw dari dokumentasi `authorizeResource` ini dipakai ketika controller berupa resource controller. Karena controller ini bukan resource controller, apakah sebaiknya `authorizeResource` dihapus saja, Kan?

https://laravel.com/docs/8.x/authorization

Reply:
Iya kang hapus aja kalau bukan authorized resource tapi gunakan authorized tapi jika dibutuhkan 😁

Reply:
Siap, Kang.. 86..
Hatur nuhun, Kang